### PR TITLE
Feature: Implement `hasPermission`

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Implement hasPermission interaction with real contracts

Task: [APP-1033](https://aragonassociation.atlassian.net/browse/APP-1033)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder of the package after the [UPCOMING] title and before the latest version.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: modules/client/src/interfaces.ts [hunk 1]
```diff
@@ -129,6 +123,14 @@ export interface IFreezePermissionDecodedParams
   permissionId: string;
 }
 
+export interface IHasPermissionParams {
+  daoAddressOrEns: string;
+  where: string;
+  who: string;
+  permission: string;
```
Comment (brickpop) [inline modules/client/src/interfaces.ts]: is this the permission name? or its hex hash?
Adding `/** doc comments */` will help here

File: modules/client/test/integration/client/methods.test.ts [hunk 2]
```diff
@@ -266,19 +271,50 @@ describe("Client", () => {
...
+          who: deployment.daoRegistry.address,
+          where: deployment.daoFactory.address,
+          permission: Permissions.REGISTER_DAO_PERMISSION,
+        };
+        const hasPermission = await client.methods.hasPermission(params);
+        // TODO
+        // this permission is granted when the contracts are deployed so it must be true
+        // it should be true?
+        expect(hasPermission).toBe(false);
+      });
```
Comment (brickpop) [inline modules/client/test/integration/client/methods.test.ts]: We should add tests for things that are actually true. For example, we should check that in a DAO with an AddressList plugin, `hasPermission()` is true when the plugin wants to execute something on the DAO

Comment (josemarinas) [reply modules/client/test/integration/client/methods.test.ts]: Yes, this is what I tried, but I was unable to find a permission that returns true, I tried with what you suggest and it is still returning false
```ts
        const params: IHasPermissionParams = {
          daoAddressOrEns: daoAddress,
          who: deployment.allowListPluginRepo.address,
          where: daoAddress,
          permission: Permissions.EXECUTE_PERMISSION,
        };

```

Comment (josemarinas) [reply modules/client/test/integration/client/methods.test.ts]: Never mind, the who and where were reversed

File: modules/client/CHANGELOG.md [hunk 3]
```diff
@@ -28,6 +28,7 @@ TEMPLATE:
 - Update Allowance estimation
 - Create dao using real contracts
 - Added `daoRegistryAddress` and `pluginRepoRegistryAddress` to web3 module
+- HasPermission using real contracts
```
Comment (mathewmeconry) [inline modules/client/CHANGELOG.md]: use `hasPermission`

**Review by mathewmeconry — APPROVED** [review]
LGTM

**Review by brickpop — APPROVED** [review]

File: modules/client/CHANGELOG.md (hunk 3 above)
Comment (brickpop) [inline modules/client/CHANGELOG.md]:
```suggestion
- `hasPermission()` using real contract calls
```

**Pull request merged**